src/stan/lang/generator.hpp

```diff
@@ -370,21 +371,30 @@ namespace stan {
         o_ << ')';
       }
 
-
       void operator()(const conditional_op& expr) const {
         bool types_prim_match
           = (expr.type_.is_primitive() && expr.type_.base_type_ == INT_T)
           || (!expr.has_var_ && expr.type_.is_primitive()
               && (expr.true_val_.expression_type()
                   == expr.false_val_.expression_type()));
+
+        std::stringstream ss;
+        if (expr.var_origin_ > local_origin) {
```

Comment:
Turn this into a function that says what it means rather than relying on order of the enum.